espei/parameter_selection/fitting_steps.py

```diff
@@ -0,0 +1,323 @@
+        ----------
+        desired_data : List[Dict[str, Any]]
+            ESPEI single phase dataset
+        fixed_model : pycalphad.Model
+            Model with all lower order (in composition) terms already fit. Pure
+            element reference state (GHSER functions) should be set to zero.
+        mole_atoms_per_mole_formula_unit : float
+            Number of moles of atoms in every mole atom unit.
+
+        Returns
+        -------
+        np.ndarray
+            Data for this feature in [qty]/mole-formula in a common reference state.
+
+        Raises
+        ------
+        ValueError
+
+        Notes
+        -----
+        pycalphad Model parameters are stored as per mole-formula quantites, but
+        the calculated properties and our data are all in [qty]/mole-atoms. We
+        multiply by mole-atoms/mole-formula to convert the units to
+        [qty]/mole-formula.
+
+        """
+        total_response = []
+        for dataset in desired_data:
+            values = np.asarray(dataset['values'], dtype=np.object_)*mole_atoms_per_mole_formula_unit
+            unique_excluded_contributions = set(dataset.get('excluded_model_contributions', []))
+            for config_idx in range(len(dataset['solver']['sublattice_configurations'])):
+                occupancy = dataset['solver'].get('sublattice_occupancies', None)
+                if dataset['output'].endswith('_FORM'):
+                    # we don't shift the reference state because we assume our
+                    # models are already in the formation reference state (by us
+                    # setting GHSERXX functions to zero explictly)
+                    pass
+                elif dataset['output'].endswith('_MIX'):
+                    if occupancy is None:
+                        raise ValueError('Cannot have a _MIX property without sublattice occupancies.')
+                    else:
+                        values[..., config_idx] += cls.transform_feature(fixed_model.models['ref'])*mole_atoms_per_mole_formula_unit
+                else:
+                    raise ValueError(f'Unknown property to shift: {dataset["output"]}')
+                for excluded_contrib in unique_excluded_contributions:
+                    values[..., config_idx] += cls.transform_feature(fixed_model.models[excluded_contrib])*mole_atoms_per_mole_formula_unit
+            total_response.append(values.flatten())
+        return total_response
+
+
+    @classmethod
+    def get_response_vector(cls, fixed_model: Model, fixed_portions: [symengine.Basic], data: [Dataset], sample_condition_dicts: [Dict[str, Any]]) -> ArrayLike:  # np.float_
+        mole_atoms_per_mole_formula_unit = fixed_model._site_ratio_normalization
+        # Define site fraction symbols that will be reused
+        phase_name = fixed_model.phase_name
+
+        # Construct flattened list of site fractions corresponding to the ravelled data (from shift_reference_state)
+        site_fractions = []
```

Comment:
I noticed lines 248-253 + 260-270 is pretty much the same as lines 152-169 where it replaces the symbols for site fractions and state variables with the actual values. Do you think it makes sense for these lines to be a separate function in FittingStep which could make the API for implementing a subclass of FittingStep a bit easier?

I'm not sure if it's very necessary to do so since a user subclassing FittingStep would likely being looking at StepHM as a reference in any case. And having these lines as a separate function may make it less clear that they are necessary if symengine symbols are introduced to data_qtys.